DataFormats/MuonDetId/src/RPCDetId.cc

```diff
@@ -167,12 +167,17 @@ RPCDetId::buildfromTrIndex(int trIndex)
   }
   trIndex = trIndex%10000;
   int sector_id = trIndex/100;
+  /*
+  // RE+1/1 :: the chamber at x=0 (phi=0) start as CH02 instead of CH01, which is not desired
+  // while for other chambers: RE+1/(2,3) and RE+2,3,4/(2,3) the rotation is arbitrary
+  // therefore commented out. Though need to understand why historically it was done like this
   if (region!=0) {
-        if ( !(ring == 1 && station > 1 && region==1)) {     
-         sector_id+=1;
-         if (sector_id==37)sector_id=1;
-     }
+    if ( !(ring == 1 && station > 1 && region==1)) {     
+      sector_id+=1;
+      if (sector_id==37)sector_id=1;
+    }
```

Comment:
Hi Yana

I printed a list of DetIds before and after changing ... The DetIds and the
associated RPC Chamber names are exactly the same.
I have run (an only 10 events) tests and the sim+digi+rechit+dqm chain
gives the same results.

As far as I understand this piece of code it introduces an arbitrarily
rotation in
RE+1/(1,2,3), RE-1/(2,3), RE+/-(2,3,4)/2,3
it seems that for rings filled with 36 chambers there is no change, while
there is a change if you only put 4 RE+1/1 chambers.

I can actually come up with a solution in which I modify that piece of code
to take out only the RE+1/1 chambers affected by this code. Although that
code of code is rather untransparent to me I ll be able to leave the code
for the existing chambers (RE+/-(1-4)/(2,3) untouched.
greets
Piet


(*) that piece of code act differently on RE+(2,3,4)/1 and RE-(2,3,4)/1,
makes me a bit suspicious... I cannot think of any asymmetry between
positive and negative endcap that would only affect RE+/-(2,3,4)/1 and no
other chambers... but it can exist.


On Fri, Apr 24, 2015 at 5:48 PM, Ianna Osborne <notifications@github.com>
wrote:

> In DataFormats/MuonDetId/src/RPCDetId.cc
> <https://github.com/cms-sw/cmssw/pull/8855#discussion_r29060113>:
>
> >    if (region!=0) {
> > -        if ( !(ring == 1 && station > 1 && region==1)) {
> > -         sector_id+=1;
> > -         if (sector_id==37)sector_id=1;
> > -     }
> > +    if ( !(ring == 1 && station > 1 && region==1)) {
> > +      sector_id+=1;
> > +      if (sector_id==37)sector_id=1;
> > +    }
>
> @pietverwilligen <https://github.com/pietverwilligen> - you are changing
> DetId numbering. Wouldn't it invalidate previously taken RPC data and
> simulation production?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/8855/files#r29060113>.
>



-- 
Piet Verwilligen

INFN -- Sezione di Bari
Via E. Orabona 4
I-70125 Bari, Italy
Phone: +39 345 74 70 642